File: 2020/d25.py
""" https://adventofcode.com/2020/day/25 """

from typing import List


def part1(data: List[int]) -> int:
    """ O(?) solution """

    pub_key_1, pub_key_2 = data

    s_num, loop_size = 1, 0
    while s_num != pub_key_1:
        s_num = (s_num * 7) % 20201227
        loop_size += 1

    encryption_key = 1
    for _ in range(loop_size):
        encryption_key = (encryption_key * pub_key_2) % 20201227

    return encryption_key


if __name__ == "__main__":
    TEST = [int(line.strip()) for line in open("tests/d25.txt", "r")]
    PUZZLE = [int(line.strip()) for line in open("puzzles/d25.txt", "r")]

    assert part1(TEST) == 14897079

    print(f"Part 1: {part1(PUZZLE)}")
    print(f"Part 2: Go and solve Jurasic Jigsaw!")

File: 2021/src/aoc_2021/d25.clj
(ns aoc-2021.d25 (:require
                  [clojure.string :as str]))

(defn- parse-input [path]
  (->> path
       (slurp)
       (str/split-lines)))

(defn part-1
  "Problem description"
  [data]
  nil)

;; Puzzle: https://adventofcode.com/2021/day/25

(def test-input (parse-input "resources/tests/d25.txt"))
(def input (parse-input "resources/puzzles/d25.txt"))

(= (part-1 test-input) nil)
(time (part-1 test-input))
